cyto_dl/nn/vits/blocks/patchify.py

```diff
@@ -99,27 +119,20 @@ def _init_weight(self):
     def get_mask(self, img, n_visible_patches, num_patches):
         B = img.shape[0]
 
-        indexes = [random_indexes(num_patches) for _ in range(B)]
+        indexes = [random_indexes(num_patches, img.device) for _ in range(B)]
         # forward indexes : index in image -> shuffledpatch
-        forward_indexes = torch.as_tensor(
-            np.stack([i[0] for i in indexes], axis=-1), dtype=torch.long
-        )
+        forward_indexes = torch.stack([i[0] for i in indexes], axis=-1)
+
         # backward indexes : shuffled patch -> index in image
-        backward_indexes = torch.as_tensor(
-            np.stack([i[1] for i in indexes], axis=-1), dtype=torch.long
-        )
+        backward_indexes = torch.stack([i[1] for i in indexes], axis=-1)
 
-        mask = torch.zeros(num_patches, B, 1)
+        mask = torch.zeros(num_patches, B, 1, device=img.device, dtype=torch.bool)
```

Comment:
use type_as instead of specifying device?

Reply:
my goal was to initialize the tensors directly on the correct device instead of initializing on cpu and then moving to the correct device